py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -473,15 +474,65 @@ def run(
         Args:
             strict_type_constraints: Usually we should set it to False unless we want to control the precision of certain layer for numeric reasons.
             algorithm_selector: set up algorithm selection for certain layer
+            tactic_sources: set up tactic sources for certain layer
         Return:
             TRTInterpreterResult
         """
+        if (
+            self.compilation_settings.custom_engine_cache is not None
+        ):  # custom_engine_cache could be None if this function is called from convert_exported_program_to_serialized_trt_engine etc.
+            if (
+                self.compilation_settings.cache_built_engines
+                or self.compilation_settings.reuse_cached_engines
+            ):
+                engine_cache = self.compilation_settings.custom_engine_cache
+                hash_val = engine_cache.get_hash(self.module)
+
+            if self.compilation_settings.reuse_cached_engines:
+                # query the cached TRT engine
+                blob = engine_cache.load(hash_val)
+                if blob is not None:  # hit the cache
+                    serialized_engine, input_names, output_names, weight_name_map = (
+                        engine_cache.unpack(blob)
+                    )
+                    self._input_names = input_names
+                    self._output_names = output_names
+                    self.weight_name_map = weight_name_map
+                    _LOGGER.info(
+                        "Found the cached engine that corresponds to this graph. It is directly loaded."
+                    )
+
+                    from torch_tensorrt.dynamo._refit import (
+                        _refit_single_trt_engine_with_gm,
+                    )
+
+                    runtime = trt.Runtime(TRT_LOGGER)
+                    engine = runtime.deserialize_cuda_engine(serialized_engine)
+
+                    _refit_single_trt_engine_with_gm(
+                        new_gm=self.module,
+                        old_engine=engine,
+                        input_list=self.input_specs,
+                        settings=self.compilation_settings,
+                        weight_name_map=weight_name_map,
+                    )
+
+                    serialized_engine = bytes(engine.serialize())
+
+                    return TRTInterpreterResult(
+                        serialized_engine,
+                        self._input_names,
+                        self._output_names,
+                        self.weight_name_map,
+                    )
```

Comment:
Here should I directly use `serialized_engine` to build `TRTInterpreterResult` or use `engine_str`:
```
with io.BytesIO() as engine_bytes:
    engine_bytes.write(serialized_engine)
    engine_str = engine_bytes.getvalue()
return TRTInterpreterResult(
    engine_str, self._input_names, self._output_names, self.weight_name_map
)
```
like line 570 below